tests/ui/js/transformsTests.js

```diff
@@ -253,32 +228,159 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
+        },
+        "test_16": {
+            input: "0",
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_17": {
+            input: "-1",
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_18": {
+            input: null,
+            expectedResult: true,
+            expectedInverseResult: false
+        },
+        "test_19": {
+            input: undefined,
+            expectedResult: undefined,
+            expectedInverseResult: undefined
+        },
+        "test_20": {
+            input: [[]],
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_21": {
+            input: NaN,
+            expectedResult: true,
+            expectedInverseResult: false
+        },
+        "test_22": {
+            input: Infinity,
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_23": {
+            input: -Infinity,
+            expectedResult: false,
+            expectedInverseResult: true
+        }
+    };
 
-        fluid.each(arrayToStringTransformTestCases, function (testCase, index) {
-            var transformSpec = $.extend({}, testCase.rules, {
-                type: "sjrk.storyTelling.transforms.arrayToString",
-                inputPath: "sourceArray"
-            });
+    /* Sets up and runs a transform through a set of test cases
+     * - "transformType": the gradeName of the transform to test
+     * - "testCases": the test cases for said transform
+     */
+    sjrk.storyTelling.transforms.runTestsForTransform = function (transformType, testCases) {
+        jqUnit.test("Test transform " + transformType, function () {
+            jqUnit.expect(Object.keys(testCases).length * 2);
 
-            var transformRules = {
-                transform: transformSpec
-            };
+            fluid.each(testCases, function (testCase, index) {
+                // merge in any extra transformation rules
+                var transformRules = {
+                    value: {
+                        transform: $.extend({}, testCase.rules, {
+                            type: transformType,
+                            inputPath: "input"
+                        })
+                    }
+                };
 
-            var resultString = fluid.model.transformWithRules(
-                { sourceArray: testCase.input },
-                { resultString: transformRules }
-            ).resultString;
-            jqUnit.assertEquals("Generated string value for test case " + index + " is as expected", testCase.expectedResult, resultString);
+                // test the transformation
+                var result = fluid.model.transformWithRules(testCase, transformRules);
+                var assertionFunction = typeof testCase.expectedResult === "object" ? jqUnit.assertDeepEq : jqUnit.assertDeepEq;
```

Comment:
Looks like you have a typo because both sides of the conditional point at `jqUnit.assertDeepEq`. However, the tests are passing because `[jqUnit.assertDeepEq](https://github.com/fluid-project/infusion/blob/c3c22dc4c3a3683280989534bb5f03f8d90d1bf9/tests/test-core/jqUnit/js/jqUnit.js#L173-L179)` will drop down to `jqUnit.assertEquals` if the expected or actual values are primitives. With that in mind, maybe you just call `jqUnit.assertDeepEq` always and skip the conditional check completely.

Reply:
Nice catch! Not sure how that one slipped by. Thanks @jobara!